src/server/zset_family.cc

```diff
@@ -2166,64 +2181,106 @@ void ZSetFamily::ZCount(CmdArgList args, const CommandContext& cmd_cntx) {
   }
 }
 
-void ZSetFamily::ZDiff(CmdArgList args, const CommandContext& cmd_cntx) {
-  vector<vector<ScoredMap>> maps(shard_set->size());
+void ZDiffInternal(CmdArgList args, const CommandContext& cmd_cntx, bool store) {
+  vector<OpResult<vector<ScoredMap>>> maps(shard_set->size(), OpStatus::SKIPPED);
+  auto* rb = static_cast<RedisReplyBuilder*>(cmd_cntx.rb);
+  const string_view dest_key = ArgS(args, 0);
+  const ShardId dest_shard = Shard(dest_key, maps.size());
```

Comment:
please use shard_set->size to compute shard id 
although maps.size and shard_size here are the save value it is more clear that the calculation of shard id a function of shard size